File: src/dataset/cpp/meetup_exercism-toemm/openAI/ChatGPT4_meetup_v1.cpp
#include "meetup.h"

#include <boost/date_time/gregorian/gregorian.hpp>

namespace meetup {

using namespace boost::gregorian;

scheduler::scheduler(months_of_year month, int year) : 
                                                        _month(month), 
                                                        _year(year), 
                                                        teenths(date(_year, _month, 13)) {}

date scheduler::monteenth() const {
    return find_teenth(Monday);
}

date scheduler::tuesteenth() const {
    return find_teenth(Tuesday);
}

date scheduler::wednesteenth() const {
    return find_teenth(Wednesday);
}

date scheduler::thursteenth() const {
    return find_teenth(Thursday);
}

date scheduler::friteenth() const {
    return find_teenth(Friday);
}

date scheduler::saturteenth() const {
    return find_teenth(Saturday);
}

date scheduler::sunteenth() const {
    return find_teenth(Sunday);
}

date scheduler::find_teenth(day_of_week day) const {
    for (date d = teenths; d.day() <= 19; d += days(1)) {
        if (d.day_of_week() == day) {
            return d;
        }
    }
    return date(); // Should never reach here
}

}  // namespace meetup

File: src/dataset/python/meetup_exercism-fruit-in/openAI/ChatGPT4_meetup_v1.py
from calendar import monthcalendar
from datetime import date

WEEK_DAYS = ['Monday', 'Tuesday', 'Wednesday',
             'Thursday', 'Friday', 'Saturday', 'Sunday']


def meetup(year, month, week, day_of_week):
    day_index = WEEK_DAYS.index(day_of_week)
    month_days = monthcalendar(year, month)

    if week == 'teenth':
        day = next(week[day_index] for week in month_days if 13 <= week[day_index] <= 19)
    elif week == 'last':
        day = next(week[day_index] for week in reversed(month_days) if week[day_index] != 0)
    else:
        week_number = int(week[0]) - 1
        try:
            day = month_days[week_number][day_index]
            if day == 0:
                raise IndexError
        except IndexError:
            raise MeetupDayException(r".+")

    return date(year, month, day)


class MeetupDayException(Exception):
    pass